glancesync/glancesync_serversfacade.py

```diff
@@ -47,36 +36,41 @@
 Therefore, this module may be changed if the API is upgraded or it is
 invoked in a different way, without affecting the main module.
 
-Current implementation works invoking the glance client through the CLI. Only
-when a functionality is not available directly from the CLI, it invokes
-the python library used by the glance and keystone clients.
+Current implementation works invoking the glance client through osclients.
+Formerly, it used to interact using a CLI wrapper.
 """
 
-# Timeout to get image list
-timeout = 30
+# Default timeout to get image list (seconds)
+_default_timeout = 30
 
 
 class ServersFacade(object):
     def __init__(self, target):
+        self.osclients = OpenStackClients(target['keystone_url'])
+        self.osclients.set_credential(target['user'], target['password'],
+                                      target['tenant'])
+        if target.get('use_keystone_v3', False):
+            self.osclients.set_keystone_version(True)
+        else:
+            self.osclients.set_keystone_version(False)
+
+        self.session = self.osclients.get_session()
+
         self.target = target
+        # This is a default value
         self.images_dir = '/var/lib/glance/images'
-        if target.get('use_keystone_v3'):
-            auth = v3.Password(
-                auth_url=target['keystone_url'], username=target['user'],
-                password=target['password'], project_name=target['tenant'],
-                project_domain_name='default', user_domain_name='default')
-            self.session = session.Session(auth=auth)
-        else:
-            auth = v2.Password(
-                auth_url=target['keystone_url'], username=target['user'],
-                password=target['password'], tenant_name=target['tenant'])
-            self.session = session.Session(auth=auth)
+        self.logger = logging.getLogger('glancesync')
+
+    def _get_glanceclient(self, region):
+        self.osclients.set_region(region)
+        return self.osclients.get_glanceclient()
 
     def get_regions(self):
         """It returns the list of regions on the specified target.
         :return: a list of region names.
         """
-        return _get_regions(self.target)
+        self.osclients.set_region(self.target)
```

Comment:
In this case the line is wrong and it is not required because the information about regions is in the credential.